homeworks/nagynorbert/8_streamlit/weather.py

```diff
@@ -0,0 +1,53 @@
+#Homework - Nagy Norbert
+import requests
+import streamlit as st
+import pandas as pd
+
+API_KEY = st.secrets["openweathermap"]["API_KEY"]
+ENDPOINT = "http://api.openweathermap.org/"
+VERSION = "2.5"
+
+@st.cache_data(ttl=86400)
+def fetch_actual_weather_data(city):
+    weather_url = f"{ENDPOINT}data/{VERSION}/weather?q={city}&appid={API_KEY}&units=metric"
+    print(f"Fetch actual weathor of {city}")
+    response = requests.get(weather_url)
+    if not city:
```

Comment:
Ez az elágazás egy kicsit ismétlődő. Ugyanazt csinálod akkor is ha nincs megadva város, illetve ha nem 200 a status code, leszámítva a warning szövegét. A return pedig mindhárom esetben ugyanazt adja vissza. Ilyen helyzetben ér lustának lenni és kevesebb kódot írni :)

```
    s_main, s_wind, coord = pd.Series(), pd.Series(), {}

    if not city:
        st.warning("City name is empty!")
    elif response.status_code != 200:
        st.warning("Data fetch was unsuccessful. City name may be incorrect!")
    else:
        resp_body = response.json()
        s_main = pd.Series(resp_body.get("main", {}))
        s_wind = pd.Series(resp_body.get("wind", {}))
        coord = resp_body.get("coord", {})
    
    return s_main, s_wind, coord
```

a 3 változót beállítjuk alapértelmezetten arra hogy minden üres. A két hibánál így elég csak a warningot megadni. Ha pedig nincs hiba akkor felülírjuk a 3 üres változót. A return pedig mivel tök ugyanaz, elég a legvégén (a 3 ágból úgyis csak 1 fut le és utána megyünk is a return-re)

Reply:
Köszönöm, ez sokkal szebb megoldás lett így.